common/js/src/popup-window/popup-opener.js

```diff
@@ -50,34 +64,48 @@ const logger = GSC.Logging.getScopedLogger('PopupWindow.PopupOpener');
 /**
  * Creates a new window.
  * @param {string} url
- * @param {!chrome.app.window.CreateWindowOptions} createWindowOptions Window
- * options.
+ * @param {!WindowOptions} windowOptions
  * @param {!Object=} opt_data Optional data to be passed to the created window.
  */
-GSC.PopupOpener.createWindow = function(url, createWindowOptions, opt_data) {
+GSC.PopupOpener.createWindow = function(url, windowOptions, opt_data) {
   const createdWindowExtends = {};
   if (opt_data !== undefined)
     createdWindowExtends['passedData'] = opt_data;
 
   goog.log.fine(
       logger,
       'Creating a popup window with url="' + url +
-          '", options=' + GSC.DebugDump.debugDump(createWindowOptions) +
+          '", options=' + GSC.DebugDump.debugDump(windowOptions) +
           ', data=' + GSC.DebugDump.debugDump(opt_data));
 
   /** @preserveTry */
   try {
     if (GSC.Packaging.MODE === GSC.Packaging.Mode.APP) {
       chrome.app.window.create(
-          url, createWindowOptions,
+          url, {
+            'alwaysOnTop': windowOptions['alwaysOnTop'],
+            'frame': windowOptions['frame'],
+            'hidden': windowOptions['hidden'],
+            'id': windowOptions['id'],
+            'innerBounds': {
+              'width': windowOptions['width'],
+            },
+            'resizable': windowOptions['resizeable'],
+            'visibleOnAllWorkspaces': windowOptions['visibleOnAllWorkspaces']
+          },
           createWindowCallback.bind(null, createdWindowExtends));
+    } else if (GSC.Packaging.MODE === GSC.Packaging.Mode.EXTENSION) {
+      chrome.windows.create({
+        url: url,
+        type: 'popup',
+        width: windowOptions['width'],
+      });
     }
```

Comment:
We need to throw an error when the `MODE` is unexpected, since otherwise `createWindow()` will do nothing, which is a surprising effect (especially if the caller then waits for a response from the window).

Reply:
you mean like a nested try/catch?